include/nuttx/can.h

```diff
@@ -326,7 +326,7 @@ struct can_frame
   uint8_t  __pad;   /* padding */
   uint8_t  __res0;  /* reserved / padding */
   uint8_t  __res1;  /* reserved / padding */
-  uint8_t  data[CAN_MAX_DLEN] aligned_data(8);
```

Comment:
why needs this change?

Reply:
> why needs this change?

this is using to fix the following build error:
CC:  src/algslib_sha256.c "comlin.c", line 189: error #1982-D: target stack alignment is insufficient to
          guarantee alignment of this variable
      struct can_frame lin_err_frame = comlin_rx_frame[chlId];
                       ^

"comlin.c", line 282: error #1982-D: target stack alignment is insufficient to
          guarantee alignment of this variable
      struct can_frame frame;
                       ^

"comlin.c", line 321: error #1982-D: target stack alignment is insufficient to
          guarantee alignment of this variable
      struct can_frame frame;

"DiagServer_IsoTp.c", line 220: error #1982-D: target stack alignment is
          insufficient to guarantee alignment of this variable
      struct canfd_frame frame = {0};

Reply:
I think this was needed as some underlying drivers expect data to be properly aligned, so upcasting to `uint32_t *` and `uint64_t *` will not lead to memfault. How did you verify that this does not break existing CAN drivers for supported devices?

Reply:
@PetervdPerk-NXP @raiden00pl @patacongo do you recall any information why CAN data field required 8 byte alignment?

Reply:
probably for Linux uapi compatibility:
https://github.com/torvalds/linux/blob/5be63fc19fcaa4c236b307420483578a56986a37/include/uapi/linux/can.h#L120-L134

Reply:
@extinguish did you test this change with any upstream SocketCan implementation ? It looks like stm32h7 and imxrt expect this data packed:
https://github.com/apache/nuttx/blob/dd7ecd9aef64b967c102808a9727a68f53025ad2/arch/arm/src/stm32h7/stm32_fdcan_sock.c#L861-L862

Reply:
but the code snippet only requires four bytes alignment.

Reply:
yes, in this case, we can add the alignment attribute to the struct level, which is supported by most compiler.